Added functionality to disable retake of quiz.

actuallyakash opened the pull request:
## Description
Admin can disable the quiz retaking ability for a student when the student gets a passed attempt. Can be enabled from the quiz builder.

## How has this been tested?
Manually.

## Types of changes
<!-- What types of changes does your code introduce?  -->
New feature

## Checklist:
- [x] This PR requires and contains at least one changelog file. <!-- To create a changelog yml file: `npm run dev changelog add -- -i` and follow the prompt. See also: https://github.com/gocodebox/lifterlms/blob/trunk/packages/dev/README.md#changelog-add -->
- [x] My code has been tested.
- [ ] My code passes all existing automated tests. <!-- Check code: `composer run-script tests-run`, Guidelines: https://github.com/gocodebox/lifterlms/blob/trunk/tests/README.md -->
- [x] My code follows the LifterLMS Coding & Documentation Standards. <!-- Check code: `composer run-script check-cs-errors`, Guidelines: https://github.com/gocodebox/lifterlms/blob/trunk/docs/coding-standards.md and https://github.com/gocodebox/lifterlms/blob/trunk/docs/documentation-standards.md -->

actuallyakash committed:
added functionality for disabling retake

after getting passed

actuallyakash committed:
added changelog

eri-trabiccolo reviewed (changes requested):
@actuallyakash 
can you please rework this PR against `dev`?
Why does it contain stuff related to the resume?

actuallyakash commented:
> @actuallyakash can you please rework this PR against `dev`? Why does it contain stuff related to the resume?

Rebased and reworked against dev. I think I had mistakenly created a branch from the resume branch.

eri-trabiccolo commented on assets/js/builder/Schemas/Quiz.js:
```diff
@@ -2,8 +2,9 @@
  * @since 3.17.6
- * @since 7.4.0 Added upsell for Question Bank and condition in `random_questions` schema.
- * @version 7.4.0
+ * @since 3.24.0 Unknown.
```
where did you take this from?

actuallyakash replied:
I rebased my branch to `dev` and got these changes. And then I added the changes that I made. Should I make a new PR?

actuallyakash replied:
I rebased my branch to dev and got these changes. Updated changelog.

eri-trabiccolo commented on assets/js/builder/Schemas/Quiz.js:
```diff
@@ -2,8 +2,9 @@
- * @since 7.4.0 Added upsell for Question Bank and condition in `random_questions` schema.
- * @version 7.4.0
+ * @since 3.24.0 Unknown.
+ * @since [version] Added upsell for Question Bank, condition in `random_questions` schema, and `disable_retake` option.
```
well, no we release a 7.4.0, so this must stay
```* @since 7.4.0 Added upsell for Question Bank and condition in `random_questions` schema.```

actuallyakash replied:
got it. Fixed now.

eri-trabiccolo commented on includes/models/model.llms.quiz.php:
```diff
@@ -51,7 +51,9 @@ class LLMS_Quiz extends LLMS_Post_Model {
 	 * Array key is the meta_key and array values is property's type.
 	 *
-	 * @var string[]
+	 * @since 3.3.0
```
where did you take this from ?

actuallyakash replied:
perhaps a wrong copy/paste. Updated now.

actuallyakash commented:
Closing this in favor of https://github.com/gocodebox/lifterlms/pull/2560

actuallyakash commented:
Closing this in favor of https://github.com/gocodebox/lifterlms/pull/2560
